hydrolib/tools/ext_old_to_new/converters.py

```diff
@@ -131,29 +190,77 @@ def convert(self, forcing: ExtOldForcing) -> Boundary:
 
         Args:
             forcing (ExtOldForcing): The contents of a single forcing block
-            in an old external forcings file. This object contains all the
-            necessary information, such as quantity, values, and timestamps,
-            required for the conversion process.
+                in an old external forcings file. This object contains all the
+                necessary information, such as quantity, values, and timestamps,
+                required for the conversion process.
+            start_time:
+                The start date of the time series data.
 
         Returns:
             Boundary: A Boundary object that represents the converted forcing
             block, ready to be included in a new external forcings file. The
             Boundary object conforms to the new format specifications, ensuring
-            compatibility with updated systems anbd models.
+            compatibility with updated systems and models.
 
         Raises:
             ValueError: If the forcing block contains a quantity that is not
             supported by the converter, a ValueError is raised. This ensures
             that only compatible forcing blocks are processed, maintaining
             data integrity and preventing errors in the conversion process.
+
+        Notes:
+            - The `root_dir` property must be set before calling this method.
+            - Since the `start_time` argument must be provided from the mdu file to convert the time series data,
+            boundary Condition can be only converted by reading the mdu file and the external forcing file is not
+            enough.
         """
+        from hydrolib.core.dflowfm.polyfile.models import PolyFile
+
+        location_file = forcing.filename.filepath
+        poly_line = forcing.filename
+        if not isinstance(poly_line, PolyFile):
+            poly_line = PolyFile(location_file)
+
+        num_files = poly_line.number_of_points
+        if self.root_dir is None:
+            raise ValueError(
+                "The 'root_dir' property must be set before calling this method."
+            )
+
+        tim_files = [
+            self.root_dir
+            / poly_line.filepath.with_name(
+                f"{poly_line.filepath.stem}_000{i + 1}.tim"
```

Comment:
Won't work for i>9. Use `_{i+1:04}` instead.

Reply:
use pathlib.Path.glob with the pli file name instead of iterating over number of points inside the pli file, as the number of points is not related to the possible number of tim files